worker/src/handlers.rs

```diff
@@ -1,13 +1,163 @@
 // Copyright (c) 2021, Facebook, Inc. and its affiliates
 // Copyright (c) 2022, Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
+use anyhow::Result;
 use async_trait::async_trait;
+use config::{SharedWorkerCache, WorkerId};
+use crypto::{NetworkPublicKey, PublicKey};
+use fastcrypto::Hash;
+use futures::{stream::FuturesUnordered, StreamExt};
+use network::{LuckyNetwork, P2pNetwork, UnreliableNetwork};
+use std::{collections::HashSet, time::Duration};
 use store::Store;
+use tap::TapFallible;
+use tokio::{
+    sync::{oneshot, watch},
+    task::JoinHandle,
+    time::{self, Timeout},
+};
+use tracing::{debug, error, trace};
 use types::{
-    error::DagError, metered_channel::Sender, Batch, BatchDigest, PrimaryToWorker,
-    PrimaryWorkerMessage, WorkerMessage, WorkerToWorker,
+    error::DagError,
+    metered_channel::{Receiver, Sender},
+    Batch, BatchDigest, PrimaryToWorker, PrimaryWorkerMessage, ReconfigureNotification,
+    WorkerBatchRequest, WorkerBatchResponse, WorkerMessage, WorkerPrimaryMessage,
+    WorkerSynchronizeMessage, WorkerToWorker,
 };
 
+#[cfg(test)]
+#[path = "tests/handlers_tests.rs"]
+pub mod handlers_tests;
+
+// Makes it possible via channels for anemo RPC handler functions to send child RPCs.
+// TODO: replace this once anemo adds support for access to the network within handlers.
+pub struct ChildRpcSender {
+    // The public key of this authority.
+    name: PublicKey,
+    // The id of this worker.
+    id: WorkerId,
+    // The worker information cache.
+    worker_cache: SharedWorkerCache,
+    // Timeout on RequestBatches RPC.
+    request_batches_timeout: Duration,
+    // Incoming RequestBatches requests to be sent out. Uses `unreliable_send` if a target is
+    // provided, or `lucky_broadcast` if a target node count is provided.
+    rx_request_batches_rpc: Receiver<(
+        Option<NetworkPublicKey>,
+        Option<usize>,
+        WorkerBatchRequest,
+        oneshot::Sender<Result<anemo::Response<WorkerBatchResponse>>>,
+    )>,
```

Comment:
nit (feel free to ignore): I think it would be beneficial to take this tuple and convert it to some named struct to make this a bit easier to wield.

Reply:
Per chat, agree it'd be better that way, but maybe skip this work if we'll be able to delete `ChildRpcSender` soonish?